packages/contracts/contracts/interfaces/CounterfactualApp.sol

```diff
@@ -21,9 +18,15 @@ interface CounterfactualApp {
     pure
     returns (bytes memory);
 
-  function resolve(bytes calldata, Transfer.Terms calldata)
+  function resolve(bytes calldata)
+    external
+    pure
+    returns (bytes memory);
+
+  // a hack
+  function resolveType()
```

Comment:
As I have mentioned above, this parameter is confusing me a bit. Could you explain it in more detail inside the comments of this pull request please?